src/components/logs/logsWebview.ts

```diff
@@ -0,0 +1,145 @@
+            currentPanel.setInfo(content, resource);
+            currentPanel.update();
+            currentPanel.panel.reveal(column);
+            return currentPanel;
+        }
+        const panel = vscode.window.createWebviewPanel(LogsPanel.viewType, "Kubernetes Logs", column || vscode.ViewColumn.One, {
+            enableScripts: true,
+
+            // And restrict the webview to only loading content from our extension's `media` directory.
+            localResourceRoots: [
+            ],
+        });
+
+        const result = new LogsPanel(panel, content, resource);
+        LogsPanel.currentPanels[resource] = result;
+        return result;
+    }
+
+    private constructor(
+        private readonly panel: vscode.WebviewPanel,
+        content: string,
+        resource: string
+    ) {
+        this.content = content;
+        this.resource = resource;
+
+        this.update();
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
+
+        this.panel.onDidChangeViewState((e: any) => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, null, this.disposables);
+    }
+
+    public setInfo(content: string, resource: string) {
+        this.content = content;
+        this.resource = resource;
+        this.update();
+    }
+
+    public dispose() {
+        delete LogsPanel.currentPanels[this.resource];
+
+        this.panel.dispose();
+
+        while (this.disposables.length) {
+            const x = this.disposables.pop();
+            if (x) {
+                x.dispose();
+            }
+        }
+    }
+
+    private update() {
+        this.panel.title = `Kubernetes logs ${this.resource}`;
+        this.panel.webview.html = `
+        <!doctype html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <title>Kubernetes logs ${this.resource}</title>
+            <script
+                src="https://code.jquery.com/jquery-3.3.1.min.js"
+                integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
+                crossorigin="anonymous"></script>
+        </head>
+        <body>
+            <div style='position: absolute; top: 15px; left: 75%'>
+                <select id='mode' style='margin-bottom: 5px' onchange='eval()'>
+                    <option value='include'>Show Matches</option>
+                    <option value='exclude'>Hide Matches</option>
+                    <option value='after'>Show logs after match</option>
+                    <option value='before'>Show logs before match</option>
+                </select>
+                <input type='text' id='regexp' onkeyup='eval()' placeholder='Regular Expression Filter' size='25'/>
+            </div>
+            <div style='margin-top: 35px'>
+                <div id='content' style="overflow-y: scroll; width: 100%; height: 100%"></div>
+            </div>
+            <script>
+              var orig = \`${this.content}\`.split('\\n');
```

Comment:
Consider doing this in the LogsPanel rather than in the embedded page.  By having all this script in the HTML string, we lose any semblance of syntax checking.  In the cluster wizard I had buttons send messages to the host, which did all the fancy logic and then sent new content back to the webview, which kept the on-page script fairly simple.

Reply:
Yes, okay